### PR TITLE
Add React Native spring animation lib (1 of 2)

### PR DESCRIPTION
**Pull request by CoolDude53**
Part of https://github.com/flutter/engine/pull/37604

Adding React Native spring animation code and license (as is). 
Part 1 of 2 per https://github.com/flutter/engine/pull/37604#issuecomment-1376536601

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/105687#issuecomment-1314305754

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by CoolDude53** [comment]
@luckysmg once this is merged, I can go ahead with the second PR

**Comment by CoolDude53** [comment]
not sure how to make the license bot happy 😬

File: ci/licenses_golden/licenses_third_party [hunk 1]
```diff
@@ -63661,4 +63661,32 @@ freely, subject to the following restrictions:
 3. This notice may not be removed or altered from any source distribution.
 ====================================================================================================
 
-Total license count: 867
+====================================================================================================
+LIBRARY: react-native
+ORIGIN: ../../../third_party/react-native/LICENSE
+TYPE: LicenseType.mit
+FILE: ../../../third_party/react-native/Libraries/Animated/animations/SpringAnimation.js
+----------------------------------------------------------------------------------------------------MIT License
```
Comment (luckysmg) [inline ci/licenses_golden/licenses_third_party:63669]: maybe just `----------------------------------------------------------------------------------------------------` ? 
Why there is a `MIT License` trailing? I didn't find this trailing in others.... 🤔

Comment (CoolDude53) [reply ci/licenses_golden/licenses_third_party:63669]: ah, that might be it, just pushed...we'll see!

**Comment by Hixie** [comment]
We'll need to update the license script. Probably easiest if I just do it for you, one sec.

**Comment by Hixie** [comment]
I got all confused by git and failed to update this PR, so instead I created a new PR: https://github.com/flutter/engine/pull/38750

Sorry about that. I'll close this one.

**Pull request closed without merging**